### PR TITLE
SAK-31705 Added 'Search Users' facility to site-info.

### PR DESCRIPTION
**Pull request by ouit0408**
Added Search input and clear search in the chef_site-siteInfo-list.vm.
If search word is in session state then Participant List is filtered for
the search term in each Participant's displayName and displayId.
Corrected the position of search box in the participant's table.

Have added new variable to STATE_SITE_PARTICIPANT_LIST in SiteAction class
to save the original site members list. Search is case insensitive.

### CODE REVIEW

**Comment by buckett** [comment]
This is how it looks in 11.
screen shot 2016-08-23 at 15 59 49

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 1]
```diff
@@ -4744,6 +4767,24 @@ private boolean coursesIntoContext(SessionState state, Context context,
...
 
+	public void doUser_search(RunData data, Context context) {
+		SessionState state = ((JetspeedRunData) data)
+				.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+
+		// read the search form field into the state object
+		String search = StringUtils.trimToNull(Validator.escapeHtml(data.getParameters().getString(FORM_SEARCH)));
+
+		// set the flag to go to the prev page on the next list
+		if (search == null) {
```
Comment (bjones86) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: `StringUtils` here; no point in searching for an empty string.

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 2]
```diff
@@ -10805,6 +10856,21 @@ private Collection getParticipantList(SessionState state) {
 		}
 
 		Collection participants = SiteParticipantHelper.prepareParticipants(siteId, providerCourseList);
+		//check for search user attribute in the state
+		String search = (String)state.getAttribute(SITE_USER_SEARCH);
+		if(search != null && (participants.size() > 0)) {
```
Comment (bjones86) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: `StringUtils` here too; no point performing the loop for an empty search string

File: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-list.vm [hunk 3]
```diff
@@ -110,8 +110,20 @@
 				utils.resizeFrame();
 			}
 		});
+		//catch enter key press for search box
+		$('#$form_search').keyup(function(event){
+			if(event.keyCode == 13){
```
Comment (bjones86) [inline site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-list.vm]: Use `===` here instead of `==`

**Comment by bjones86** [comment]
Minor comments in-line; otherwise LGTM

**Comment by buckett** [comment]
@ouit0408 Are you able to fix the comments on this one?

**Comment by ouit0408** [comment]
it is in my to-do list, will fix.

**Comment by ouit0408** [comment]
implemented review comments and PR is updated.

**Comment by buckett** [comment]
Thanks.

**Pull request merged**